### PR TITLE
Remove Istio 1.17 version check

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/security/authz/model/principal.go [hunk 1]
```diff
@@ -57,16 +57,7 @@ func principalNot(principal *rbacpb.Principal) *rbacpb.Principal {
 	}
 }
 
-func principalAuthenticated(name *matcher.StringMatcher, useAuthenticated bool) *rbacpb.Principal {
```
Comment (hzxuzhonghu) [inline pilot/pkg/security/authz/model/principal.go:60]: cc @kyessenov I think this is safe, unless there are some other caveats

Comment (hzxuzhonghu) [reply pilot/pkg/security/authz/model/principal.go]: restored `useAuthenticated`

File: pilot/pkg/networking/plugin/authz/authorization.go [hunk 2]
```diff
@@ -49,9 +48,7 @@ func NewBuilder(actionType ActionType, push *model.PushContext, proxy *model.Pro
 		IsCustomBuilder: actionType == Custom,
 	}
 	policies := push.AuthzPolicies.ListAuthorizationPolicies(proxy.ConfigNamespace, proxy.Labels)
-	if !util.IsIstioVersionGE117(proxy.IstioVersion) {
-		option.UseAuthenticated = true
-	}
+	option.UseAuthenticated = false
```
Comment (howardjohn) [inline pilot/pkg/networking/plugin/authz/authorization.go:51]: if this is always `false` not sure we need an option here.

Although probably the grpcgen should use NewBuilder...

Comment (hzxuzhonghu) [reply pilot/pkg/networking/plugin/authz/authorization.go]: removed UseAuthenticated

**Review by howardjohn — APPROVED** [review]

**Pull request merged**